controllers/helper_test.go

```diff
@@ -538,6 +540,52 @@ func testBuildManagedZone(name, ns, domainName string) *kuadrantdnsv1alpha1.Mana
 	}
 }
 
+// testEndpointsTraversable consumes an array of endpoints and returns a boolean
+// indicating presence of that path from host to all IPs
+func testEndpointsTraversable(endpoints []*endpoint.Endpoint, host string, ips []string) bool {
+	for _, ip := range ips {
+		allTargetsFound := false
+		for _, ep := range endpoints {
+			// the target on the endpoint exists as an endpoint
+			if ep.DNSName == host {
+				// target is our IP
+				if slices.Contains(ep.Targets, ip) {
+					return true
+				} else {
+					// target is not out IP. Use target as a host and check for existence of Endpoints with such a DNSName
+					for _, target := range ep.Targets {
+						// if at least one returns as true allTargetsFound will be locked in true
+						allTargetsFound = allTargetsFound || testEndpointsTraversable(endpoints, target, []string{ip})
```

Comment:
Probably no point calling `testEndpointsTraversable` unless the endpoint is a CNAME type

Reply:
if `endpoint.Target` is an IP it will just check it it `if ep.DNSName == IP` and happily fail. Otherwise there should be a regex check of some sort to tell if that is IP or not :thinking: I don't have any strong feelings about one way or another. Let me see what I can do

Reply:
I'm not saying it won't work, it's just pointless so why do it. If you hit an A record with multiple targets it will call that function x amount of times for no reason. You don't need to do regexp either,  since you already know the type of endpoint `ep.RecordType`

Reply:
Changed. Now should be more obvious